python/mxnet/gluon/block.py

```diff
@@ -1029,14 +1029,35 @@ def _build_cache(self, *args):
 
         arg_dict, aux_dict = dict(), dict()
         if self._backend:
-            ctx = args[0].context
+            # set context for inputs
+            _, _, ctx_set, _ = _gather_type_ctx_info(list(args))
+            ctx = ctx_set.pop() if len(ctx_set) > 0 else None
```

Comment:
why compute the whole `ctx_set` when we are only using the first element?

Reply:
we need to see if any args have a context, i guess we dont support mixed mx.sym.var and NDArrays in the same call (ie. has to be all of one or the other). but i was just following suit with how other Gluon APIs do things.